### PR TITLE
Conventions for Arcade repo

### PR DESCRIPTION
**Pull request by JohnTortugo**
Closes: https://github.com/dotnet/arcade/issues/148

Only included what have been discussed so far.

### CODE REVIEW

File: Documentation/ArcadeConventions.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Arcade Conventions
+
+These are the conventions used in the Arcade repository.
+
+### Naming of Scripts
+
+File names and file extensions should be lowercase. For instance: `build.cmd`, `build.sh`, `restore.cmd`, etc.
```
Comment (weshaggard) [inline Documentation/ArcadeConventions.md]: Can you please list out the entire rule set from the issue?

- Filenames should be all lowercase build.cmd or nuget.config. Only exceptions are for files that needs to be cased a particular way for an existing set of tools to read them (Example: Directory.Build.props which msbuild expects exactly that case on linux).
- Filenames with multiple works should use kebab-casing like init-tools.sh.
- Msbuild based targets and props files for a particular library should match the exact casing of the library package they belong to, like Microsoft.DotNet.Build.Tasks.Feed.targets.

File: Documentation/ArcadeConventions.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Arcade Conventions
+
+These are the conventions used in the Arcade repository.
+
+### Naming of Scripts
```
Comment (weshaggard) [inline Documentation/ArcadeConventions.md]: This is the file naming convention not just for scripts.

File: Documentation/ArcadeConventions.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+File names and file extensions should be lowercase. For instance: `build.cmd`, `build.sh`, `restore.cmd`, etc.
+
+### Dependent Packages Version
+
```
Comment (maririos) [inline Documentation/ArcadeConventions.md]: We could specify here how to do that (similar to what you did in PR #374 )

File: Documentation/ArcadeConventions.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+Package versions are stored in properties on the `eng\Versions.props` file. Use those properties to include the correct version of the package. New properties will be included as needed.
```
Comment (maririos) [inline Documentation/ArcadeConventions.md]: maybe a link to the version.props file?

File: Documentation/ArcadeConventions.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+If your project depend on a package which is also part of the .NET SDK used by Arcade (check `global.json`) the project should use the version of the package available in the SDK. Otherwise, the latest stable version of the package should be used.
```
Comment (maririos) [inline Documentation/ArcadeConventions.md]: Could you provide an example? that might help to clarify this

**Comment by JohnTortugo** [comment]
@maririos / @weshaggard I addressed your comments. Thanks.

**Comment by maririos** [comment]
Thanks for writing this Cesar!

**Pull request merged**